stdlib/src/collections/dict.mojo

```diff
@@ -177,15 +178,26 @@ struct _DictValueIter[
         V: The value type of the elements in the dictionary.
         dict_mutability: Whether the reference to the vector is mutable.
         dict_lifetime: The lifetime of the List
+        forward: The iteration direction. `False` is backwards.
     """
 
     alias ref_type = Reference[V, dict_mutability, dict_lifetime]
 
-    var iter: _DictEntryIter[K, V, dict_mutability, dict_lifetime]
+    var iter: _DictEntryIter[K, V, dict_mutability, dict_lifetime, forward]
 
     fn __iter__(self) -> Self:
         return self
 
+    fn __reversed__[
```

Comment:
**Suggestion** Do you mind adding a changelog entry [here](https://github.com/modularml/mojo/blob/fd304d2f3f60a51816b8672cafdea41da35bd456/docs/changelog.md) for your work?  Adding an entry under the "New" section and crediting yourself and this PR link would be great!